### PR TITLE
atuin: add disable key options

### PR DESCRIPTION
**Pull request by Janik-Haag**
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

### CODE REVIEW

**Comment by janalemke** [comment]
Thanks for writing this! I was just about to try and add this myself when I saw your PR. My only (nitpicky) comment is that the short descriptions for the options are wrong in so far as the options disable the keybinds for all shells and not only zsh.

**Comment by ncfavier** [comment]
I'd rather add a generic `shellIntegrationFlags` option with type `listOf str`.

**Comment by ncfavier** [comment]
It could get more in the future.

**Comment by Janik-Haag** [comment]
Okay I will implement that in a few hours.

**Comment by Janik-Haag** [comment]
@janalemke @ncfavier would be nice if you could review again :)

File: modules/programs/atuin.nix [hunk 1]
```diff
@@ -50,6 +50,15 @@ in {
       '';
     };
 
+    flags = mkOption {
+      default = [ ];
+      type = types.listOf types.str;
+      example = literalExpression "[ \"--disable-up-arrow\" \"--disable-ctrl-r\" ]";
```
Comment (ncfavier) [inline modules/programs/atuin.nix:56]:
```suggestion
      example = [ "--disable-up-arrow" "--disable-ctrl-r" ];
```

File: modules/programs/atuin.nix [hunk 2]
```diff
@@ -50,6 +50,15 @@ in {
...
+      description = ''
+        Configure flags to append to the shell hook
```
Comment (ncfavier) [inline modules/programs/atuin.nix:58]:
```suggestion
        Flags to append to the shell hook.
```

File: modules/programs/atuin.nix [hunk 3]
```diff
@@ -78,7 +87,11 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
+  config = let
+    disableKeys = ""
+      + optionalString (cfg.flags != [ ]) " " 
+      + optionalString (cfg.flags != [ ]) (concatStringsSep " " cfg.flags);
```
Comment (ncfavier) [inline modules/programs/atuin.nix:93]:
```suggestion
    flagsStr = escapeShellArgs cfg.flags;
```

File: modules/programs/atuin.nix [hunk 4]
```diff
@@ -91,18 +104,18 @@ in {
     programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
       if [[ :$SHELLOPTS: =~ :(vi|emacs): ]]; then
         source "${pkgs.bash-preexec}/share/bash/bash-preexec.sh"
-        eval "$(${cfg.package}/bin/atuin init bash)"
+          eval "$(${cfg.package}/bin/atuin init bash${disableKeys})"
```
Comment (ncfavier) [inline modules/programs/atuin.nix:107]:
```suggestion
          eval "$(${cfg.package}/bin/atuin init bash ${flagsStr})"
```

File: modules/programs/atuin.nix [hunk 5]
```diff
@@ -91,18 +104,18 @@ in {
...
       fi
     '';
 
     programs.zsh.initExtra = mkIf cfg.enableZshIntegration ''
       if [[ $options[zle] = on ]]; then
-        eval "$(${cfg.package}/bin/atuin init zsh)"
+        eval "$(${cfg.package}/bin/atuin init zsh${disableKeys})"
```
Comment (ncfavier) [inline modules/programs/atuin.nix:113]:
```suggestion
        eval "$(${cfg.package}/bin/atuin init zsh ${flagsStr})"
```

File: modules/programs/atuin.nix [hunk 6]
```diff
@@ -91,18 +104,18 @@ in {
...
       fi
     '';
 
     programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration ''
-      ${cfg.package}/bin/atuin init fish | source
+      ${cfg.package}/bin/atuin init fish${disableKeys} | source
```
Comment (ncfavier) [inline modules/programs/atuin.nix:118]:
```suggestion
      ${cfg.package}/bin/atuin init fish ${flagsStr} | source
```

**Comment by Janik-Haag** [comment]
I'm pushing the changes when I figured out what causes the test cases to fail (no it's not the trailing white space)

File: modules/programs/atuin.nix [hunk 7]
```diff
@@ -91,18 +101,18 @@ in {
     programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
       if [[ :$SHELLOPTS: =~ :(vi|emacs): ]]; then
         source "${pkgs.bash-preexec}/share/bash/bash-preexec.sh"
-        eval "$(${cfg.package}/bin/atuin init bash)"
+          eval "$(${cfg.package}/bin/atuin init bash ${flagsStr})"
```
Comment (ncfavier) [inline modules/programs/atuin.nix:104]:
```suggestion
        eval "$(${cfg.package}/bin/atuin init bash ${flagsStr})"
```

**Review by ncfavier — APPROVED** [review]

**Pull request merged**